yacl/crypto/primitives/tpre/tpre_test.cc

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 Chengfang Financial Technology Co., Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "yacl/crypto/primitives/tpre/tpre.h"
+
+#include "absl/strings/escaping.h"
+#include "gtest/gtest.h"
+
+namespace yacl::crypto::test {
+
+class TpreTest : public testing::Test {};
+
+TEST_F(TpreTest, Test1) {
+  std::string iv = "123456789";
+  TPRE tpre;
+  Keys keys;
+
+  /************************* Phase 1 *************************/
+  // Start testing encryption and decryption functions
+  std::unique_ptr<EcGroup> ecc_group = EcGroupFactory::Create("sm2");
+
+  std::pair<std::unique_ptr<Keys::PublicKey>, std::unique_ptr<Keys::PrivateKey>>
+      key_pair_A = keys.GenerateKeyPair(std::move(ecc_group));
+
+  Keys::PublicKey* pk_A = key_pair_A.first.get();
+  Keys::PrivateKey* sk_A = key_pair_A.second.get();
```

Comment:
Please use smart pointers instead of raw pointers

Reply:
Revised according to comments